server/data/data-api/src/main/java/org/apache/james/dlp/api/DLPRulesStore.java

```diff
@@ -0,0 +1,41 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.dlp.api;
+
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.james.core.Domain;
+
+public interface DLPRulesStore {
+
+    Map<DLPRuleId, DLPRule> retrieveRules(Domain domain);
+
+    Optional<DLPRule> retrieveRule(Domain domain, DLPRuleId dlpRuleId);
+
+    DLPRuleId store(Domain domain, DLPRule rule);
```

Comment:
we cannot allow to store only one Rule, all rules must be stored at once, else you need to handle ordering

Reply:
Why would you need ordering? Isn't it a AND between rules & AND commutes?

Reply:
I'm not sure it's absolutely needed. But once you get a list of rules, you can be tempted to order them. For example for performance reasons.